src/CommandIsolator.cpp

```diff
@@ -134,44 +141,30 @@ process::Future<ContainerLimitation> CommandIsolatorProcess::watch(
 
 process::Future<::mesos::ResourceStatistics> CommandIsolatorProcess::usage(
     const ContainerID& containerId) {
-  if (m_usageCommand.isNone()) {
-    ::mesos::ResourceStatistics stats;
-    stats.set_timestamp(Clock::now().secs());
-    return stats;
-  }
+  if (m_usageCommand.isNone()) return emptyStats();
 
   logging::Metadata metadata = {containerId.value(), "usage"};
 
   JSON::Object inputsJson;
   inputsJson.values["container_id"] = JSON::protobuf(containerId);
 
-  Try<string> output = CommandRunner(m_isDebugMode, metadata)
-                           .run(m_usageCommand.get(), stringify(inputsJson));
-
-  if (output.isError()) {
-    LOG(WARNING) << "Unable to parse output: " << output.error();
-    ::mesos::ResourceStatistics stats;
-    stats.set_timestamp(Clock::now().secs());
-    return stats;
-  }
-
-  if (output->empty()) {
-    ::mesos::ResourceStatistics stats;
-    stats.set_timestamp(Clock::now().secs());
-    return stats;
-  }
-
-  Result<::mesos::ResourceStatistics> resourceStatistics =
-      jsonToProtobuf<::mesos::ResourceStatistics>(output.get());
-
-  if (resourceStatistics.isError()) {
-    LOG(WARNING) << "Unable to deserialize ResourceStatistics: "
-                 << resourceStatistics.error();
-    ::mesos::ResourceStatistics stats;
-    stats.set_timestamp(Clock::now().secs());
-    return stats;
-  }
-  return resourceStatistics.get();
+  return CommandRunner(m_isDebugMode, metadata)
+      .asyncRun(m_usageCommand.get(), stringify(inputsJson))
+      .then([](Try<string> output) -> Future<::mesos::ResourceStatistics> {
+        if (output.isError() || output->empty()) {
+          LOG(WARNING) << "Unable to parse output: " << output.error();
```

Comment:
Wont work if output is not an error, you should have dedicated messages

Reply:
You're right, it was actually crashing, and this has been fixed